Add contributor documentation about naming diagnostic IDs (#1559)
apple/swift-docc: Sources/SwiftDocC/SwiftDocC.docc/SwiftDocC/Adding-Diagnostics.md

it _can_ make sense to use the old diagnostic identifier as the group identifier.
This way, developers' custom severity configurations for the previous diagnostic identifier continues to apply to the collection of more specific diagnostics.

> Note: 
> Both individual diagnostic identifiers and diagnostic group identifiers are considered user-facing information.

> > Use a very terse but descriptive word or phrase of uppercase words joined together, for example `MultipleTopicsSections` and avoid internal development jargon.

Reviewer: ```suggestion
> Use a very terse but descriptive word or phrase of title-cased words joined together, for example `MultipleTopicsSections` and avoid internal development jargon.
```

Author: Nice catch. I think I meant to use "capitalized words" here.
```suggestion
> Use a very terse but descriptive word or phrase of capitalized words joined together, for example `MultipleTopicsSections` and avoid internal development jargon.
```

> > The identifier only needs to communicate which issues it's referring to so that developers can configure the diagnostic's severity. 

Reviewer: Question: can a single identifier refer to multiple issues?

Author: Yes. A single identifier can refer to multiple diagnostics. This would mean that if the developer configures the severity of that identifier, it applies to all those diagnostics that use that identifier.

Alternatively, multiple diagnostics with different individual identifiers can specify the same _group_ identifier to still allow the developer to control their severity together via the group identifier.

> > The identifier only needs to communicate which issues it's referring to so that developer's can configure the diagnostic's severity. 

Reviewer 2: ```suggestion
> The identifier only needs to communicate which issues it's referring to so that developers can configure the diagnostic's severity. 
```

> > The identifier's purpose _isn't_ to make the developer understand what the issue is about. 

Reviewer: Nitty nit: italics and bold both emphasise a word in different ways, I feel bold might be more appropriate here, along with expanding the contraction to make it more obvious:

```suggestion
> The identifier's purpose __is not__ to make the developer understand what the issue is about. 
```

> That responsibility belongs to the ``Diagnostic/summary`` and the ``Diagnostic/explanation``.
> 

> > Diagnostic identifiers with a "org.swift.docc." prefix _won't_ be displayed to the developer (the diagnostic is presented without its identifier). 

Reviewer: Nitty nit: same as above

```suggestion
> Diagnostic identifiers with a "org.swift.docc." prefix __will not__ be displayed to the developer (the diagnostic is presented without its identifier). 
```

> > These identifiers aren't considered user-facing yet and changing those identifiers _isn't_ considered a breaking change.

Reviewer: Nitty nit: same as above

```suggestion
> These identifiers aren't considered user-facing yet and changing those identifiers __is not__ considered a breaking change.
```

> > This gives us the opportunity to go through any existing diagnostics one by one and update them to based on our most up-to-date understanding of what makes a clear and actionable diagnostic.

Reviewer 2: ```suggestion
> This gives us the opportunity to go through any existing diagnostics one by one and update them based on our most up-to-date understanding of what makes a clear and actionable diagnostic.
```
